components/raftstore-v2/src/batch/store.rs

```diff
@@ -405,14 +406,17 @@ impl<EK: KvEngine, ER: RaftEngine, T> StorePollerBuilder<EK, ER, T> {
             if prefix == SPLIT_PREFIX {
                 file_system::remove_dir_all(&path)?;
                 continue;
-            }
-            if prefix.is_empty() {
+            } else if prefix == MERGE_IN_PROGRESS_PREFIX || prefix == MERGE_SOURCE_PREFIX {
```

Comment:
Turns out `MERGE_SOURCE_PREFIX` should be handled before L401, otherwise if source is destroyed, the checkpoint can be lost.